util/stmtsummary/stmtevicted_summary.go

```diff
@@ -0,0 +1,123 @@
+package stmtsummary
+
+import (
+	"container/list"
+	"github.com/pingcap/parser/mysql"
+	"github.com/pingcap/tidb/types"
+	"github.com/pingcap/tidb/util/hack"
+	"github.com/pingcap/tidb/util/kvcache"
+	"strconv"
+	"sync"
+	"time"
+)
+
+var StmtEvictedMap = newstmtEvictedMap()
+
+type stmtEvictedByDigestMap struct {
+	// It's rare to read concurrently, so RWMutex is not needed.
+	sync.Mutex
+	summaryMap *kvcache.SimpleLRUCache
+	// beginTimeForCurInterval is the begin time for current summary.
+	beginTimeForCurInterval int64
+}
+
+type stmEvictedByDigestKey struct {
+	hash          []byte
+	beginTimeHash string
+}
+
+type stmEvictedByDigest struct {
+	sync.Mutex
+	beginTime int64
+	endTime int64
+	isInit	bool
+	history	*list.List
+	Count	int64
+	historyMax int
+}
+
+func newstmtEvictedMap() *stmtEvictedByDigestMap {
+	return &stmtEvictedByDigestMap{
+		//Only record 96 time window(48H)
+		//maybe we can modify this later
+		summaryMap:              		kvcache.NewSimpleLRUCache(96,0,0),
+		beginTimeForCurInterval:		900,
```

Comment:
Don't use const.
The capacity should keep consistant with the config of `stmtSummaryByDigestMap `.

Reply:
👌🏻